File: Modules/Common/Resources/SIE.Web.Resources/ShiftTypes/Commands/SetDefaultCommand.cs
using SIE.Resources.ShiftTypes;
using SIE.Web.Command;
using System;

namespace SIE.Web.Resources.ShiftTypes.Commands
{
    /// <summary>
    /// 默认命令
    /// </summary>
    [JsCommand("SIE.Web.Resources.ShiftTypes.Commands.SetDefaultCommand")]
    public class SetDefaultCommand : ViewCommand
    {
        /// <summary>
        /// 默认命令设置
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var shiftType = args.Data.ToJsonObject<ShiftType>();
            if (null == shiftType)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(shiftType)));
            }
            RT.Service.Resolve<ShiftTypeController>().SetDefaultShiftType(shiftType);
            return true;
        }
    }
}

File: Modules/Common/Resources/SIE.Web.Resources/ShiftTypes/Commands/SetDefaultCommand.js
SIE.defineCommand('SIE.Web.Resources.ShiftTypes.Commands.SetDefaultCommand', {
    meta: { text: "设置缺省", group: "edit", iconCls: "icon-ListConfig icon-blue" },

    canExecute: function (view) {
        if (view.getSelection() == null || view.getSelection().length !== 1) {
            return false;
        } 
        if (view.getCurrent() !== null && view.getCurrent().data.IsDefault !== 0)
            return false;
        if (view.getCurrent().isNew())
            return false;
        return true;
    },

    execute: function (view, source) {
        var entity = view.getCurrent();
        if (entity) {
            entity = entity.data;
            var msg = Ext.String.format('是否将班制{0}设置为缺省？'.t(), entity.Name);
            SIE.Msg.askQuestion(msg, function () {
                view.execute({
                    data: entity,
                    success: function (res) {
                        view.loadData();
                    }
                }, view);
            });
        }
    }
});